hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/ethereum/EthTxData.java

```diff
@@ -322,4 +266,113 @@ EthTxData replaceTo(byte[] to) {
                 r,
                 s);
     }
+
+    /**
+     * Encodes the transaction data into a EthTxData according to legacy RLP format.
+     *
+     * @return the encoded transaction data
+     */
+    private static EthTxData populateLegacyEthTxData(RLPItem rlpItem, byte[] rawTx) {
+        List<RLPItem> rlpList = rlpItem.asRLPList().elements();
+        if (rlpList.size() != 9) {
+            return null;
+        }
+
+        byte[] chainId = null;
+        byte[] v = rlpList.get(6).asBytes();
+        BigInteger vBI = new BigInteger(1, v);
+        byte recId = vBI.testBit(0) ? (byte) 0 : 1;
+        if (vBI.compareTo(BigInteger.valueOf(34)) > 0) {
+            chainId = vBI.subtract(BigInteger.valueOf(35)).shiftRight(1).toByteArray();
+        }
+
+        return new EthTxData(
```

Comment:
I guess I don't get why this stuff has to appear twice, sort of.  Here and in `encodeTx` above.

Also, as an aside, I guess we don't have to touch this file too often, which is good, because there has to be a better way to do these finicky details, matching external specs - the various EIPs - with nothing to tie this code to them except comments about which field is which.  Only reason this is ok (IMO) is that it _works_ in real life against real data.